File: app/src/main/java/com/example/neon/deardiary/settings/SettingsActivity.java
package com.example.neon.deardiary.settings;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import com.example.neon.deardiary.R;
import com.example.neon.deardiary.data.source.DataSource;
import com.example.neon.deardiary.data.source.local.originaldb.DiaryLocalDataSource;
import com.example.neon.deardiary.util.ActivityUtils;

public class SettingsActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_settings);
        SettingsFragment settingsFragment = (SettingsFragment) getFragmentManager().findFragmentById(R.id.settings_container);
        if(settingsFragment==null){
            settingsFragment = new SettingsFragment();
            ActivityUtils.addFragmentToActivity(getFragmentManager(),settingsFragment,R.id.settings_container);
        }
        DataSource dataSource = DiaryLocalDataSource.getInstance(this);
        //create the presenter
        new SettingsPresenter(dataSource,settingsFragment);
    }
}

File: app/src/main/java/com/example/neon/deardiary/settings/SettingsActivity.kt
package com.example.neon.deardiary.settings

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.example.neon.deardiary.R
import com.example.neon.deardiary.data.source.local.originaldb.DiaryLocalDataSource
import com.example.neon.deardiary.utils.ActivityUtil

class SettingsActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_settings)
        var settingsFragment: SettingsFragment? = fragmentManager.findFragmentById(R.id.settings_container) as? SettingsFragment
        if (settingsFragment == null) {
            settingsFragment = SettingsFragment()
            ActivityUtil.addFragmentToActivity(fragmentManager, settingsFragment, R.id.settings_container)
        }
        val dataSource = DiaryLocalDataSource.getInstance(this)
        //create the presenter
        SettingsPresenter(dataSource, settingsFragment)
    }
}
